lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -551,6 +557,20 @@ def is_online_index_pointer(self):
     def source_info(self):
         return self._source_info
 
+    def validate_version(self):
+        if not self.is_scm:
+            version_req = "A SemVer-compliant version or '*' is required. See https://semver.org to learn how to compose it correctly. "
+        else:
+            version_req = f"A string version is required. Got {self.ver} ({type(self.ver)})."
+        version_err = f"Invalid version found for the collection '{self}'. {version_req}"
+        if not isinstance(self.ver, string_types):
+            raise AnsibleError(version_err)
```

Comment:
Can we use specific exceptions for things raised in the internals and only convert them into `AnsibleError` on the outer layer?
A suitable exception type for this case could be a `TypeError`, for example.

Reply:
Okay. This was just refactoring the existing exception, but I can look at raising a TypeError instead.

Reply:
Fixed.